_posts/2019-09-19-intellij-bazel-sync.md

```diff
@@ -0,0 +1,538 @@
+
+We will also explore the process logs, and connect the dots between them and the
+generated files to crystallize our understanding.
+
+Let's dive in!
+
+## Structure of `.ijwb`
+
+Using a [Spring Boot project
+example](https://github.com/bazelbuild/rules_jvm_external/tree/master/examples/spring_boot),
+[BlazeSyncManager](https://github.com/bazelbuild/intellij/blob/master/base/src/com/google/idea/blaze/base/sync/BlazeSyncManager.java)
+generates this directory on a project sync.
+
+```
+$ tree -a .ijwb/
+.ijwb/
+├── .bazelproject
+├── .blaze
+│   ├── libraries
+│   ├── modules
+│   │   ├── .project-data-dir.iml
+│   │   └── .workspace.iml
+│   └── remoteOutputCache
+└── .idea
+    ├── .name
+    ├── codeStyles
+    │   └── codeStyleConfig.xml
+    ├── externalDependencies.xml
+    ├── libraries
+    │   ├── Runner_deploy_ijar_d8363141.xml
+    │   ├── hamcrest_library_1_3_c425095b.xml
+    │   ├── spring_beans_5_1_5_RELEASE_c7c017d3.xml
+    │   ├── spring_boot_2_1_3_RELEASE_f01f6f60.xml
+    │   ├── spring_boot_autoconfigure_2_1_3_RELEASE_77fee7ea.xml
+    │   ├── spring_boot_starter_web_2_1_3_RELEASE_968a1469.xml
+    │   ├── spring_boot_test_2_1_3_RELEASE_1631a67.xml
+    │   ├── spring_boot_test_autoconfigure_2_1_3_RELEASE_9cfb3ab1.xml
+    │   ├── spring_context_5_1_5_RELEASE_69b9cfff.xml
+    │   ├── spring_core_5_1_5_RELEASE_b1cbb181.xml
+    │   ├── spring_test_5_1_5_RELEASE_be462134.xml
+    │   └── spring_web_5_1_5_RELEASE_90feac64.xml
+    ├── misc.xml
+    ├── modules.xml
+    ├── runConfigurations.xml
+    ├── vcs.xml
+    └── workspace.xml
+```
+
+`.ijwb` ,or `.aswb`/`.clwb` for Android Studio and CLion respectively, is known
+as the **project directory**. It contains metadata about the project that
+bridges Bazel and IntelliJ project models.
+
+Let's investigate the components of this directory individually.
+
+## Project view file
+
+```
+$ tree -a
+.
+├── .bazelproject 
+```
+
+This is the IJwB [project view file
+](https://ij.bazel.build/docs/project-views.html) containing project-wide
+settings, like targets to sync, Bazel flags, and enabled languages. Check this
+file into your project's version control to share Bazel project settings. For
+example, the basic project view file looks like this:
+
+```
+directories:
+  # Add the directories you want added as source here
+  # By default, we've added your entire workspace ('.')
+  .
+
+# Automatically includes all relevant targets under the 'directories' above
+derive_targets_from_directories: true
+
+targets:
+  # If source code isn't resolving, add additional targets that compile it here
+
+additional_languages:
+  # Uncomment any additional languages you want supported
+  # android
+  # dart
+  # kotlin
+  # python
+  # scala
+```
+
+## Bazel data subdirectory
+
+`.ijwb/.blaze` is the Bazel data subdirectory, containing mostly IntelliJ module
+definitions.
+
+Note that the actual persistent serialized data is not in this directory. To
```

Comment:
Much of the persisted serialized data in this directory -- might be worth rephrasing this.

Reply:
done. thanks!